src/Microsoft.TemplateEngine.Utils/DictionaryExtensions.cs

```diff
@@ -25,5 +46,32 @@ public static class DictionaryExtensions
                 return cloneDict;
             }
         }
+
+        /// <summary>
+        /// Adds a content of given dictionary to current dictionary.
+        /// </summary>
+        /// <typeparam name="TKey"></typeparam>
+        /// <typeparam name="TValue"></typeparam>
+        /// <param name="dict">Dictionary to receive another values.</param>
+        /// <param name="another">Dictionary to be merged into current.</param>
+        /// <param name="conflictingKeysResolution">Way of resolving keys conflicts.</param>
+        /// <exception cref="Exception">Thrown if key is already present in current dictionary and <see cref="ConflictingKeysResolution.Throw"/> strategy was requested.</exception>
+        public static void Add<TKey, TValue>(
+            this IDictionary<TKey, TValue> dict,
+            IReadOnlyDictionary<TKey, TValue> another,
+            ConflictingKeysResolution conflictingKeysResolution = ConflictingKeysResolution.Overwrite)
+        {
+            foreach (var pair in another)
+            {
+                if (conflictingKeysResolution == ConflictingKeysResolution.Overwrite || !dict.ContainsKey(pair.Key))
+                {
+                    dict[pair.Key] = pair.Value;
+                }
+                else if (conflictingKeysResolution == ConflictingKeysResolution.Throw)
+                {
+                    throw new Exception($"Key [{pair.Key}] is already present.");
```

Comment:
Currently no exceptions in `Microsoft.TemplateEngine.Utils` are localized, plus this code path is not currently exercised by our code - so skipped localization for now.

Does that sound acceptable? Or should we add localization to Utils as well?

Reply:
It's better to have them localized. Also `InvalidOperationException` or `ArgumentException` might be better in this case, or may be custom exception.  .NET throws `ArgumentException` in similar case.